src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/FundingModal/FundingModal.tsx

```diff
@@ -0,0 +1,131 @@
+import { Wallet } from '@phosphor-icons/react';
+import React, { useState, type FC } from 'react';
+
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { ActionTypes } from '~redux';
+import { type FundExpenditurePayload } from '~redux/sagas/expenditures/fundExpenditure.ts';
+import { formatText } from '~utils/intl.ts';
+import { type SelectOption } from '~v5/common/Fields/Select/types.ts';
+import ActionButton from '~v5/shared/Button/ActionButton.tsx';
+import Button from '~v5/shared/Button/Button.tsx';
+import Modal from '~v5/shared/Modal/index.ts';
+
+import DecisionMethodSelect from '../DecisionMethodSelect/index.ts';
+
+import {
+  fundingDecisionMethodDescriptions,
+  fundingDecisionMethodItems,
+} from './consts.ts';
+import TokenItem from './TokenItem.tsx';
+import { type FundingModalProps } from './types.ts';
+
+const FundingModal: FC<FundingModalProps> = ({
+  onClose,
+  expenditure,
+  ...rest
+}) => {
+  const [method, setMethod] = useState<SelectOption['value']>();
+  const { colony } = useColonyContext();
+
+  const fundingItems = Object.entries(
+    expenditure.slots.reduce((grouped, slot) => {
+      if (!slot.payouts) return grouped;
+
+      const mergedPayouts = { ...grouped };
+
+      slot.payouts.forEach((payout) => {
+        const { tokenAddress, amount, networkFee } = payout;
+
+        mergedPayouts[tokenAddress] = {
+          amount:
+            (mergedPayouts[tokenAddress]?.amount || 0) + parseFloat(amount),
```

Comment:
Those operations should use `BigNumber` to avoid running into JS precision problems